lib/iris/_lazy_data.py

```diff
@@ -135,3 +135,64 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def convert_nans_array(array, nans=None, result_dtype=None):
+    """
+    Convert a :class:`~numpy.ndarray` that may contain one or more NaN values
+    to either a :class:`~numpy.ma.core.MaskedArray` or a
+    :class:`~numpy.ndarray` with the NaN values filled.
+
+    Args:
+
+    * array:
+        The :class:`~numpy.ndarray` to be converted.
+
+    Kwargs:
+
+    * nans:
+        If `nans` is None, then raise an exception if the `array` contains
+        any NaN values (default).
```

Comment:
> nans is None is never used 

But it is [here](https://github.com/bjlittle/iris/blob/0b82b11a7a1abcd9f55e1218d2f3ca2a046ab394/lib/iris/_lazy_data.py#L182) ?